### ISSUE TITLE
WSL prevents Xilinx ISE running on windows.

### ISSUE THREAD

**Issue by prime6809** [microsoft/WSL#9284]
### Version

Microsoft Windows [Version 10.0.19045.2311]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Distro Version

Ubuntu 22.04.1

### Repro Steps

Windows v 22H2, with all windows updates applied.
Xilinx webpack installed on WINDOWS.
WSL installed from Microsoft store.
WSL is enabled in Control Panel->Programs and Features->Turn windows features on or off 

Xilinx tools (ise.exe, impact.exe) will not launch, they hang during launching and can only be killed from task manager.
Disabling WSL in control panel allows them to launch successfully.

This only seems to happen once update KB5020030 is installed, before this update WSL and Xilinx programs worked together correctly.

### Expected Behavior

Xilinx tools launch correctly.

### Actual Behavior

Hang during launch of Xilinx programs.

**Comment by pmartincic** [comment]
Can you take logs while reproducing this?
/logs

**Comment by prime6809** [comment]
[WslLogs-2022-12-14_18-15-20.zip](https://github.com/microsoft/WSL/files/10230592/WslLogs-2022-12-14_18-15-20.zip)

Logs whilst triggering the problem.

Cheers.

Phill.

**Comment by asthecryflows** [comment]
I stumbled upon this github issue after experiencing this problem myself. I found an explanation and solution here, which I figured I might as well share here to potentially save some headaches for anyone else who finds this page first.
https://www.exxosforum.co.uk/forum/viewtopic.php?p=95884#p95884

The gist is that libportability.dll (one of xilinx's) barfs when enumerating network shares once it gets to the \\wsl shares.
